### ISSUE TITLE
DBusConnection - RejectedExecutionException

### ISSUE THREAD

**Issue by swiesend** [swiesend/secret-service#21]
There are problems with versions [1.2.0](https://github.com/swiesend/secret-service/pull/19)-[1.2.3](https://github.com/swiesend/secret-service/tree/v1.2.3) on shutting down the static initial connection to check whether there is a secret-service or not.

```
Exception in thread "DBusConnection" java.util.concurrent.RejectedExecutionException: Task org.freedesktop.dbus.connections.AbstractConnection$3@6f96a320 rejected from java.util.concurrent.ThreadPoolExecutor@231c97bd[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2055)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:825)
	at java.base/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1355)
	at org.freedesktop.dbus.connections.AbstractConnection.executeInWorkerThreadPool(AbstractConnection.java:946)
	at org.freedesktop.dbus.connections.AbstractConnection.handleMessage(AbstractConnection.java:920)
	at org.freedesktop.dbus.connections.AbstractConnection.handleMessage(AbstractConnection.java:709)
	at org.freedesktop.dbus.connections.IncomingMessageThread.run(IncomingMessageThread.java:43)
```

**Comment by swiesend** [comment]
And there is also:

```

[Thread-2] ERROR org.freedesktop.secret.handlers.SignalHandler - java.util.concurrent.RejectedExecutionException: Task org.freedesktop.dbus.connections.AbstractConnection$1@5b44a31c rejected from java.util.concurrent.ThreadPoolExecutor@2f01c4f[Shutting down, pool size = 1, active threads = 0, queued tasks = 0, completed tasks = 26]
[Thread-2] ERROR org.freedesktop.secret.handlers.SignalHandler - Could not remove all signal handlers from the D-Bus.
```

**Comment by swiesend** [comment]
```
[main] INFO org.freedesktop.secret.ServiceTest - isRemote()
Exception in thread "DBusConnection" java.util.concurrent.RejectedExecutionException: Task org.freedesktop.dbus.connections.AbstractConnection$3@5a0fc9cb rejected from java.util.concurrent.ThreadPoolExecutor@1f0d4cee[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2057)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:827)
	at java.base/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1357)
	at org.freedesktop.dbus.connections.AbstractConnection.executeInWorkerThreadPool(AbstractConnection.java:946)
	at org.freedesktop.dbus.connections.AbstractConnection.handleMessage(AbstractConnection.java:920)
	at org.freedesktop.dbus.connections.AbstractConnection.handleMessage(AbstractConnection.java:709)
	at org.freedesktop.dbus.connections.IncomingMessageThread.run(IncomingMessageThread.java:43)
```

**Comment by infeo** [comment]
I believe this is a race condition, e.g. the DBus object is already disconnected, but still need to handle some messages. After analyzing this isssue I believe the error lies within the dbus-java project. For a more in depth analysis, have a look at the linked issue.

**Comment by infeo** [comment]
@swiesend can you try to reproduce the issue with the current HEAD of dbus-java? The maintainer applied some changes that possibly fix the problem.

**Comment by swiesend** [comment]
Wow, @infeo that is some very in depth bug tracking! Thank you for writing the issue, which I ignored way too long and feel guilty about of not having it communicated it better to you guys.

So I tested it with the current HEAD of `dbus-java` and it seems **not** to fix the `RejectedExecutionException` (1/50 times) and neither the blocked shutdown (2/50 times with a duration of about 5 min) of 50 graceful cryptomator shutdowns.

I tested with a local `cryptomator` `1.6.0-SNAPSHOT` using the new integrations-api `integrations-linux` `0.2.0-SNAPSHOT` depending on my `secret-service` [1.2.4-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/de/swiesend/secret-service/1.2.4-SNAPSHOT).

with `dbus-java` HEAD on:
```git


commit c1ee1213ce8aefc33bddeee3567dc1ae3b2c235d (HEAD -> master, origin/master, origin/HEAD)
Author: David M <maniac@case-of.org>
Date:   Wed Nov 11 17:39:03 2020 +0100

    Implemented some of the suggestions from #123
```

```
sebastian@home:~/develop/github/cryptomator/cryptomator/main$ buildkit-linux/launcher-linux.sh
23:33:36.899 [main] DEBUG org.cryptomator.common.Environment - java.library.path: /usr/local/cuda-11.1/lib64:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
23:33:36.903 [main] DEBUG org.cryptomator.common.Environment - user.language: en
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - user.region: null
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - logback.configurationFile: null
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - cryptomator.settingsPath: ~/.config/Cryptomator/settings.json
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - cryptomator.ipcPortPath: ~/.config/Cryptomator/ipcPort.bin
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - cryptomator.keychainPath: null
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - cryptomator.logDir: ~/.local/share/Cryptomator/logs
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - cryptomator.mountPointsDir: ~/.local/share/Cryptomator/mnt
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - cryptomator.minPwLength: null
23:33:36.904 [main] DEBUG org.cryptomator.common.Environment - cryptomator.buildNumber: null
23:33:36.942 [main] DEBUG org.cryptomator.common.ShutdownHook - Registered shutdown hook.
23:33:37.017 [main] DEBUG org.cryptomator.common.settings.SettingsProvider - Attempting to load settings from /home/sebastian/.config/Cryptomator/settings.json
23:33:37.066 [main] WARN org.cryptomator.common.settings.SettingsJsonAdapter - Unsupported vault setting found in JSON: keychainBackend
23:33:37.066 [main] INFO org.cryptomator.common.settings.SettingsProvider - Settings loaded from /home/sebastian/.config/Cryptomator/settings.json
23:33:37.747 [main] INFO  org.cryptomator.launcher.Cryptomator - Starting Cryptomator 1.6.0-SNAPSHOT on Linux 5.4.0-53-generic (amd64)
23:33:37.987 [JavaFX Application Thread] INFO  o.c.ui.launcher.FxApplicationStarter - JavaFX Runtime started.
Exception in thread "DBusConnection" java.util.concurrent.RejectedExecutionException: Task org.freedesktop.dbus.connections.AbstractConnection$3@76756d94 rejected from java.util.concurrent.ThreadPoolExecutor@573f8321[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2057)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:827)
	at java.base/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1357)
	at org.freedesktop.dbus.connections.AbstractConnection.executeInWorkerThreadPool(AbstractConnection.java:946)
	at org.freedesktop.dbus.connections.AbstractConnection.handleMessage(AbstractConnection.java:920)
	at org.freedesktop.dbus.connections.AbstractConnection.handleMessage(AbstractConnection.java:709)
	at org.freedesktop.dbus.connections.IncomingMessageThread.run(IncomingMessageThread.java:43)
23:33:41.800 [App Background Thread 003] INFO  o.c.ui.unlock.UnlockWorkflow - Unlock of 'sebastian' succeeded.
23:33:41.918 [DBus Worker Thread-2] INFO  o.f.secret.handlers.SignalHandler - Received signal Collection.ItemChanged: /org/freedesktop/secrets/collection/login/478
23:33:41.937 [DBus Worker Thread-3] INFO  o.f.secret.handlers.SignalHandler - Received signal Collection.ItemChanged: /org/freedesktop/secrets/collection/login/478
23:33:45.495 [JavaFX Application Thread] INFO  o.cryptomator.ui.fxapp.UpdateChecker - Current version: 1.6.0-SNAPSHOT, lastest version: 1.5.8
```


```
sebastian@home:~/develop/github/cryptomator/cryptomator/main$ buildkit-linux/launcher-linux.sh 
23:23:17.440 [main] DEBUG org.cryptomator.common.Environment - java.library.path: /usr/local/cuda-11.1/lib64:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - user.language: en
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - user.region: null
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - logback.configurationFile: null
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - cryptomator.settingsPath: ~/.config/Cryptomator/settings.json
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - cryptomator.ipcPortPath: ~/.config/Cryptomator/ipcPort.bin
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - cryptomator.keychainPath: null
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - cryptomator.logDir: ~/.local/share/Cryptomator/logs
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - cryptomator.mountPointsDir: ~/.local/share/Cryptomator/mnt
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - cryptomator.minPwLength: null
23:23:17.445 [main] DEBUG org.cryptomator.common.Environment - cryptomator.buildNumber: null
23:23:17.492 [main] DEBUG org.cryptomator.common.ShutdownHook - Registered shutdown hook.
23:23:17.573 [main] DEBUG org.cryptomator.common.settings.SettingsProvider - Attempting to load settings from /home/sebastian/.config/Cryptomator/settings.json
23:23:17.626 [main] WARN org.cryptomator.common.settings.SettingsJsonAdapter - Unsupported vault setting found in JSON: keychainBackend
23:23:17.626 [main] INFO org.cryptomator.common.settings.SettingsProvider - Settings loaded from /home/sebastian/.config/Cryptomator/settings.json
23:23:18.250 [main] INFO  org.cryptomator.launcher.Cryptomator - Starting Cryptomator 1.6.0-SNAPSHOT on Linux 5.4.0-53-generic (amd64)
23:23:18.510 [JavaFX Application Thread] INFO  o.c.ui.launcher.FxApplicationStarter - JavaFX Runtime started.
23:23:22.532 [App Background Thread 003] INFO  o.c.ui.unlock.UnlockWorkflow - Unlock of 'sebastian' succeeded.
23:23:22.642 [DBus Worker Thread-2] INFO  o.f.secret.handlers.SignalHandler - Received signal Collection.ItemChanged: /org/freedesktop/secrets/collection/login/478
23:23:22.665 [DBus Worker Thread-3] INFO  o.f.secret.handlers.SignalHandler - Received signal Collection.ItemChanged: /org/freedesktop/secrets/collection/login/478
23:23:24.858 [JavaFX Application Thread] INFO  o.cryptomator.ui.quit.QuitController - Locked sebastian
23:23:24.867 [main] INFO  org.cryptomator.launcher.Cryptomator - UI shut down

^C


^C
[testing-dbus-3.2.4-SNAPSHOT-c1ee1213ce8aefc33bddeee3567dc1ae3b2c235d.log](https://github.com/swiesend/secret-service/files/5533704/testing-dbus-3.2.4-SNAPSHOT-c1ee1213ce8aefc33bddeee3567dc1ae3b2c235d.log)

^C^C^C
```
